constructor-manager/src/constructor_manager/utils/shortcuts.py

```diff
@@ -0,0 +1,184 @@
+    """Create or remove shortcuts for a package and version.
+
+    Parameters
+    ----------
+    package : str
+        The name of the package.
+    version : str
+        The version of the package.
+    remove_action : bool
+        If True, remove the shortcuts. If False, create the shortcuts.
+
+    Returns
+    -------
+    paths : list
+        The paths to the shortcuts that were created.
+    """
+    if target_prefix is None:
+        target_prefix = get_prefix_by_name(f"{package}-{version}")
+
+    base_prefix = get_prefix_by_name("base")
+
+    # TODO: Is this standard for all platforms?
+    metadata_path = target_prefix / "Menu" / f"{package}-menu.json"
+    func = remove if remove_action else install
+    try:
+        paths = func(
+            metadata_path,
+            target_prefix=target_prefix,
+            base_prefix=base_prefix,
+        )
+    except Exception:
+        paths = []
+
+    return paths
+
+
+def create_shortcut(package, version, target_prefix=None):
+    """Create a shortcut for a package and version.
+
+    Parameters
+    ----------
+    package : str
+        The name of the package.
+    version : str
+        The version of the package.
+
+    Returns
+    -------
+    paths : list
+        The paths to the shortcuts that were created.
+    """
+    return _create_remove_shortcut(
+        package, version, remove_action=False, target_prefix=target_prefix
+    )
+
+
+def remove_shortcut(package, version, target_prefix=None):
+    """Remove a shortcut for a package and version.
+
+    Parameters
+    ----------
+    package : str
+        The name of the package.
+    version : str
+        The version of the package.
+
+    Returns
+    -------
+    paths : list
+        The paths to the shortcuts that were removed.
+    """
+    return _create_remove_shortcut(
+        package, version, remove_action=True, target_prefix=target_prefix
+    )
+
+
+def open_application(package, version, target_prefix=None):
+    """Open the application using the shortcuts created by menuinst.
+
+    Parameters
+    ----------
+    package : str
+        The name of the package.
+    version : str
+        The version of the package.
+    target_prefix : str, optional
+        The target prefix to use. If not provided, the default is used.
+
+    Returns
+    -------
+    ret_code : int
+        The return code from the command that opened the application.
+    """
+    paths = _locations(package, version, target_prefix=target_prefix)
+    if paths:
+        path = paths[0]
+
+    plat = platform.system()
+    if plat == "Darwin":
+        ret_code = os.system(f"open '{path}'")
+    elif plat == "Linux":
+        cmd = ""
+        with open(path) as fh:
+            for line in fh:
+                if line.startswith("Exec="):
+                    cmd = line.split("=", 1)[1].strip()
+        if cmd:
+            ret_code = os.system(cmd)
```

Comment:
😳

I shouldn't be surprised there's no better way to do this. Apparently `gtk-launch` should do it but obviously not on every system.

Reply:
Yes, this is a bit of a "weird part" of the code. Used some suggestions from @jaimergp work on menuinst